PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -513,6 +514,23 @@ LRESULT ScintillaEditView::scintillaNew_Proc(HWND hwnd, UINT Message, WPARAM wPa
 		{
 			break;
 		}
+
+		case WM_RBUTTONDOWN:
+		{
+			int clickX = GET_X_LPARAM(lParam);
+			int clickY = GET_Y_LPARAM(lParam);
+			int clickPos = static_cast<int>(execute(SCI_POSITIONFROMPOINT, clickX, clickY));
+			int posX = static_cast<int>(execute(SCI_POINTXFROMPOSITION, clickPos));
+			if (clickX >= posX)
+			{
+				// if right-click in the editing area (not the margins!),
+				// don't let this go to Scintilla because it will 
+				// move the caret to the right-clicked location,
+				// cancelling any selection made by the user
+				return TRUE;
+			}
+			break;
+		}
 	}

```

Comment:
The function itself is good, it behaves like WIndows-Notepad now.

I did some experiments to find out why it works, because it looked strange to me. _Why is the text position needed to find out the margin width?_

The `execute(SCI_POINTXFROMPOSITION, clickPos)` is actually wrong. The Scintilla doc says:
`SCI_POINTXFROMPOSITION(<unused>, position pos) → int`.

So the correct call would be `execute(SCI_POINTXFROMPOSITION, 0, clickPos)`. Which wrecks the function.

This code works, because the `execute(SCI_POINTXFROMPOSITION, clickPos)` always returns the x pixel position of the text position 0, which is equal to the margin width.

So the same effect can be reached with less code lines:

		case WM_RBUTTONDOWN:
		{
			int clickX = GET_X_LPARAM(lParam);
			int marginX = static_cast<int>(execute(SCI_POINTXFROMPOSITION, 0, 0));
			if (clickX >= marginX)
			{
				. . .